src/Cake.ProGet/Asset/ProGetConfigurationExtensions.cs

```
@@ -38,5 +38,19 @@ internal static void Apply(this ProGetConfiguration config, HttpWebRequest reque
                 request.Timeout = (int)config.RequestTimeout.Value.TotalMilliseconds;
             }
         }
+
+        internal static HttpClient CreateClient(this ProGetConfiguration config)
```

I think this method (and the subsequent invocations of `.Apply` in the caller classes) create a mutually exclusive scenario that we'd want to avoid.   i.e., a caller could provide the following and see unexpected behavior.
```
new ProGetConfiguration 
{
  ProGetUser = "user",
  ProGetPassword = "password",
  UseDefaultCredentials = true
}
```

It might be better to refactor the extension class to be a more legitimate factory implementation wherein we can instantiate a well-formed `HttpWebRequest` or `HttpClient` object with the correct attributes applied.  More to the point, if we run into an unexpected combination of configuration values, as above, we can throw an appropriate exception and fail the operation in an expected way.  It (the factory class) would also be inherently testable.